EngineLayer/MetaMorpheusEngine.cs

```diff
@@ -80,6 +80,55 @@ public static double CalculatePeptideScore(MsDataScan thisScan, List<MatchedFrag
             return score;
         }
 
+        public static double CalculateAllChargesPeptideScore(MsDataScan thisScan, List<MatchedFragmentIon> matchedFragmentIons)
```

Comment:
This seems like a separate PR, but if we're trying to score all charges, shouldn't that happen in the matched fragment ion method? Those fragment ions with multiple charges could then be scored using the normal scoring method. Guess I'm not sure why this method is needed.

Reply:
Normal search only looks for one match ion for one fragment, and if it accepts it  then it doesn't try to look for different charge states of that same fragment. The score will be the number of matched ions and plus some fraction calculated by intensity(matchedFragmentIons[i].Intensity / thisScan.TotalIonCurrent).
Like b1,b2,b3 will have score 3.xxx;
But when generating library, we need look for match ions with all charges. So we will have b1,b2,b3, b1^2, b2^3. If using the normal scoring function, the score will be 5.xxxx, which is not proper in my opinion. I think the score for b1 and b1^2 should also be 1 plus some some fraction calculated by intensity, because they are matching the same fragment ion just with different charges. So b1,b2,b3, b1^2, b2^3 should be also 3.xxx(but a little higher than b1,b2,b3 as the fraction part)  rather than 5.xxx.

I hope this answers your question. And please tell me if any of my thoughts are not proper.

Reply:
That makes sense. Can you add a header to the method so that it's clear how this method is different?